analysers/analyser_osmosis_highway_area_access.py

```diff
@@ -91,6 +91,58 @@
     NOT barrier.waytags->'{vehicle}' IN ('no', 'use_sidepath', 'unknown')
 """
 
+sql31 = """
+SELECT
+  highways.id,
+  barrier.id,
+  ST_AsText(barrier.geom)
+FROM
+  highways
+  JOIN nodes AS barrier ON
+    barrier.tags != ''::hstore AND
+    barrier.id = ANY(highways.nodes) AND
+    --barrier.tags?'barrier' AND -- commented to prevent slow route
+    barrier.tags->'barrier' NOT IN ('border_control', 'cattle_grid', 'entrance', 'height_restrictor', 'toll_booth') AND
+    NOT (barrier.tags?'access' OR barrier.tags?'access:conditional') AND
+    NOT (barrier.tags?'vehicle' OR barrier.tags?'vehicle:conditional') AND
+    NOT (barrier.tags?'motor_vehicle' OR barrier.tags?'motor_vehicle:conditional')
+WHERE
+  highways.level <= 3 AND
+  NOT (highways.tags?'access' OR highways.tags?'access:conditional') AND
+  NOT (highways.tags?'vehicle' OR highways.tags?'vehicle:conditional') AND
+  NOT (highways.tags?'motor_vehicle' OR highways.tags?'motor_vehicle:conditional')
+"""
+
+
+sql41 = """
+SELECT
+  ways.id,
+  barrier.id,
+  ST_AsText(barrier.geom)
+FROM
+  highways AS ways
+  JOIN way_nodes ON
+    way_nodes.way_id = ways.id
```

Comment:
Usually join on linestring index is faster ways.linestring && minor_highway.linestring. Have you try it ?

Reply:
Good one, I forgot that linestrings also intersect if they just share 1 node (=crossing). Will try later this week!